### ISSUE TITLE
ngDialog.open should handler promise error

### ISSUE THREAD

**Issue by any35** [likeastore/ngDialog#401]
The following promise could not be resolve when the the template couldn't be loaded. all the dialog's promise(closePromise) are dead.
```
 $q.all({
   template: loadTemplate(options.template || options.templateUrl),
   locals: $q.all(resolve)
}).then(...)
```
Is there any workaround to handle exception in ngDialog.open.

**Comment by faceleg** [comment]
Could you provide a more complete example in JSFiddle?

**Comment by any35** [comment]
Here it is,
 http://jsfiddle.net/15Lqnduw/   
Thank you.

**Comment by faceleg** [comment]
You're right, I see this is not ideal. How do you suggest it should be improved?
